enterprise/api/v1/views.py

```diff
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
+            errors = []
+            enrolled_count = 0
+            license_uuids = serializer.validated_data.get('license_info')
+            course_run_keys = serializer.validated_data.get('courses')
+            users_missing_licenses = []
+
+            # Get a list of user emails from the license info.
+            emails = set(license_uuids.keys())
+            try:
+                for email in emails:
+                    try:
+                        validate_email_to_link(email, enterprise_customer, raise_exception=False)
+                        if len(license_uuids[email]) != len(course_run_keys):
+                            users_missing_licenses.append(email)
+                    except ValidationError as error:
+                        errors.append(error)
+            except ValidationError as exc:
+                errors.append(exc)
+
+            if users_missing_licenses:
+                msg = 'All requested users must have a valid license for each of the courses requested to be ' \
+                      'enrolled in. Users missing licenses: {}'.format(users_missing_licenses)
+                return Response(msg, status=HTTP_400_BAD_REQUEST)
+
+            if errors:
+                return Response(errors, status=HTTP_400_BAD_REQUEST)
+
+            for email in emails:
+                models.EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+
+            if list(emails):
+                for course_run_key, mode in course_run_keys.items():
+                    discount = serializer.validated_data.get('discount', 0.0)
+                    enrollment_reason = serializer.validated_data.get('reason')
+                    succeeded, pending, _ = enroll_users_in_course(
+                        enterprise_customer=enterprise_customer,
+                        course_id=course_run_key,
+                        course_mode=mode,
+                        emails=emails,
+                        enrollment_requester=request.user,
+                        enrollment_reason=enrollment_reason,
+                        discount=discount,
+                        sales_force_id=serializer.validated_data.get('salesforce_id'),
+                        license_uuids=license_uuids
+                    )
+                    enrolled_count = len(succeeded + pending)
+                    if serializer.validated_data.get('notify'):
+                        enterprise_customer.notify_enrolled_learners(
+                            catalog_api_user=request.user,
+                            course_id=course_run_key,
+                            users=succeeded + pending,
+                        )
+                    self._create_ecom_orders_for_enrollments(
+                        course_run_key,
+                        mode,
+                        discount,
+                        serializer.validated_data.get('salesforce_id'),
+                        succeeded,
+                    )
+            return Response(
```

Comment:
You need to be tracking failed as you go around the loop here same as you are successes.